pkg/validate/validate.go

```diff
@@ -221,8 +219,13 @@ func validateFQDN(p *api.Properties) (errs []error) {
 	if p == nil {
 		errs = append(errs, fmt.Errorf("masterProfile cannot be nil"))
 	}
-	if !isValidHostname(p.FQDN) {
-		errs = append(errs, fmt.Errorf("invalid properties.fqdn %q", p.FQDN))
+	// check if FQDN is not empty, and validate format if not.
+	if p.FQDN == "" {
+		errs = append(errs, fmt.Errorf("properties.fqdn cannot be empty"))
+	} else {
+		if !isValidHostname(p.FQDN) {
+			errs = append(errs, fmt.Errorf("invalid properties.fqdn %q", p.FQDN))
+		}
```

Comment:
This prevents both validation errors. `""` is one errors `()` is other.

Reply:
simplify to `if p.FQDN == "" || !isValidHostname(p.FQDN) {`